File: Backend/Nodejs/src/routes/song.ts
import { Router } from "express";
import multer from "multer";
import { SongController } from "../controllers/song";
import { s3Storage } from "../libs/s3client";

export const songRouter = Router();
const uploadMP3 = multer({ storage: s3Storage({ userType: "song" }) });

songRouter.get("/", SongController.getSongs);
songRouter.get("/:id", SongController.getSong);
songRouter.post(
  "/newSong",
  uploadMP3.fields([
    { name: "cover", maxCount: 1 },
    { name: "source", maxCount: 1 },
  ]),
  SongController.createSong,
);
songRouter.patch(
  "/",
  uploadMP3.fields([
    { name: "cover", maxCount: 1 },
    { name: "source", maxCount: 1 },
  ]),
  SongController.editSong,
);
songRouter.delete("/:id", SongController.deleteSong);

File: Backend/Python/src/routes/song.py
from flask import Blueprint
from src.controllers.song import SongController

song_blueprint = Blueprint("blueprint", __name__)

song_blueprint.route("/", methods=["GET"])(SongController.get_songs)
song_blueprint.route("/<int:id>/<email>", methods=["GET"])(SongController.get_song)
song_blueprint.route("/newSong", methods=["POST"])(SongController.create_song)
song_blueprint.route("/<int:id>", methods=["DELETE"])(SongController.delete_song)
song_blueprint.route("/", methods=["PATCH"])(SongController.edit_song)
